### PR TITLE
docs: add comment for method Redirect and Header

### PR DESCRIPTION
**Pull request by Tom-debug110**
#### What type of PR is this?

docs: Documentation only changes

#### Check the PR title.

- [x] This PR title match the format: `<type>(optional scope): <description>`.
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [ ] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io).

#### (Optional) Translate the PR title into Chinese.

补充 RequestContext 的Redirect 和Header 方法的注释,以 Gin 中对应的方法注释为模版,降低理解成本

#### (Optional) Which issue(s) this PR fixes:
[#867](https://github.com/cloudwego/hertz/issues/867)

### CODE REVIEW

**Review by li-jin-gou — DISMISSED** [review]

File: pkg/app/context.go [hunk 1]
```diff
@@ -769,10 +769,16 @@ func (ctx *RequestContext) Reset() {
 	ctx.conn = nil
 }
 
+// Redirect returns an HTTP redirect to the specific location.
+// Note that this will not stop the current handler.
+// in other words, if existed,the following code maybe execute and cause unexpected result
```
Comment (welkeyever) [inline pkg/app/context.go:774]: `following code`有点抽象，可以补充说明一下，如果想要立即终止后续handler/中间件调用，可以配合:
```
ctx.Abort()
return
```
另外，最后一句缺个句号（同下

Comment (welkeyever) [reply pkg/app/context.go:774]: Hi @Tom-debug110 , is there any update recently?

Comment (Tom-debug110) [reply pkg/app/context.go]: sorry,I am busy with campus recruitment recently, thank you for your opinion.

Comment (welkeyever) [reply pkg/app/context.go]: Maybe use spaces and newlines to improve readability like this：https://github.com/cloudwego/hertz/pull/925/files

To separate the code in the commnet.

Comment (Tom-debug110) [reply pkg/app/context.go]: ok. Done.

File: pkg/app/context.go [hunk 2]
```diff
@@ -769,10 +769,20 @@ func (ctx *RequestContext) Reset() {
 	ctx.conn = nil
 }
 
+// Redirect returns an HTTP redirect to the specific location.
+// Note that this will not stop the current handler.
+// in other words, even if Redirect() is called, the remaining handlers will still be executed and cause unexpected result.
```
Comment (welkeyever) [inline pkg/app/context.go:774]: in -> In

File: pkg/app/context.go [hunk 3]
```diff
@@ -769,10 +769,20 @@ func (ctx *RequestContext) Reset() {
...
+// So it should call Abort to ensure the remaining handlers for this request are not called.
```
Comment (welkeyever) [inline pkg/app/context.go:775]: "handlers for this request" -> "handlers of this request"
&&
"are not called." -> "will not be called."

File: pkg/app/context.go [hunk 4]
```diff
@@ -769,10 +769,20 @@ func (ctx *RequestContext) Reset() {
...
+//
+//	ctx.Abort()
+//	return
 func (ctx *RequestContext) Redirect(statusCode int, uri []byte) {
 	ctx.redirect(uri, statusCode)
 }
 
+// Header is an intelligent shortcut for ctx.Response.Header.Set(key, value).
+// It writes a header in the response.
+// If value == "", this method removes the header `ctx.Response.Header.Del(key)`
```
Comment (welkeyever) [inline pkg/app/context.go:785]: needs a "." at the end.

**Review by welkeyever — APPROVED** [review]
LGTM, thx~

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**